code/modules/spells/spell_types/ethereal_jaunt.dm

```diff
@@ -75,6 +93,22 @@
 						break
 		REMOVE_TRAIT(target, TRAIT_IMMOBILIZED, type)
 
+/**
+ * Updates the exit point of the jaunt
+ *
+ * Called when the jaunting mob holder moves, this updates the backup exit-jaunt
+ * location, in case the jaunt ends with the mob still in a wall. Five
+ * spots are kept in the list, in case the last few changed since we passed
+ * by (doors closing, engineers building walls, etc)
+ */
+/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/proc/update_exit_point(mob/living/target)
+	SIGNAL_HANDLER
+	var/turf/location = get_turf(target)
+	if(location.is_blocked_turf(ignore_climbable = TRUE))
+		return
+	exit_point_list[5] = null
+	exit_point_list.Insert(1, location)
```

Comment:
```suggestion
	exit_point_list.Insert(1, location)
	if(length(exit_point_list) >= 5)
		exit_point_list.Cut(5)
```
This will avoid the list growing and holding `null` values.